cli/formatter/format.go

```diff
@@ -0,0 +1,794 @@
+	})
+	var columnConfigs []table.ColumnConfig
+	for i := 1; i <= colAmount; i++ {
+		columnConfigs = append(columnConfigs,
+			table.ColumnConfig{
+				Number:      i,
+				Transformer: colWidthTransformer,
+				WidthMax:    opts.ColWidthMax,
+			},
+		)
+	}
+	t.SetColumnConfigs(columnConfigs)
+}
+
+// constructMarkdownTable performs construct table in markdown notation.
+func constructMarkdownTable(markdownTable []string, colAmount int,
+	rowsAmount int, opts Opts) string {
+	if !opts.TransposeMode {
+		markdownTable = insertStrToStrSlice(markdownTable, 0,
+			constructMarkdownEmptyRow(colAmount))
+		markdownTable = insertStrToStrSlice(markdownTable, 1,
+			constructMarkdownHeaderSeparatorRow(colAmount))
+	} else {
+		markdownTable = insertStrToStrSlice(markdownTable, 0,
+			constructMarkdownEmptyRow(rowsAmount))
+		markdownTable = insertStrToStrSlice(markdownTable, 1,
+			constructMarkdownHeaderSeparatorRow(rowsAmount))
+	}
+	var res string
+	for _, v := range markdownTable {
+		res = res + v + "\n"
+	}
+
+	return res
+}
+
+// encodeToCellJsonVal performs encoding as json values for table cell.
+func encodeToCellJsonVal(rowColVal interface{}) (string, error) {
+	jsonVal, err := json.Marshal(convertMapIntrfToMapStr(rowColVal))
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsonVal), nil
+}
+
+// encodeScalarNodeToTableRow performs encoding scalar renderNode to table row.
+func encodeScalarNodeToTableRow(val renderNode) interface{} {
+	var row interface{}
+	switch val.(type) {
+	case float64, float32:
+		row = strconv.FormatFloat(val.(float64), 'f', -1, 64)
+	case nil:
+		row = "nil"
+	default:
+		row = fmt.Sprintf("%v", val)
+	}
+
+	return row
+}
+
+// renderMapsWithEqualKeys returns maps with equal keys as single table string.
+func renderMapsWithEqualKeys(maps []map[string]renderNode, opts Opts) (string, error) {
+	var commonKeys []string
+	for mapKey := range maps[0] {
+		commonKeys = append(commonKeys, mapKey)
+	}
+
+	sort.Strings(commonKeys)
+
+	t := table.NewWriter()
+	var rows []table.Row
+	rows = append(rows, constructHeaderRowForMap(commonKeys))
+
+	for _, mapVal := range maps {
+		var rowVals table.Row
+		for _, commonKey := range commonKeys {
+			if getRenderNodeType(mapVal[commonKey]) == scalarRenderNodeType {
+				rowVals = append(rowVals, encodeScalarNodeToTableRow(mapVal[commonKey]))
+			} else {
+				// Encoding as json value for map/array table cell cases.
+				jsonVal, err := encodeToCellJsonVal(mapVal[commonKey])
+				if err != nil {
+					return "", err
+				}
+				rowVals = append(rowVals, jsonVal)
+			}
+		}
+		rows = append(rows, rowVals)
+	}
+
+	t.Style().Options.SeparateRows = true
+	var rowsAmount = len(rows)
+	if opts.TransposeMode {
+		rows = transposeRows(rows)
+	}
+	t.AppendRows(rows)
+
+	if opts.NoGraphics {
+		t.SetStyle(table.Style{Box: StyleWithoutGraphics})
+	}
+	if opts.ColWidthMax > 0 {
+		handleColWidth(len(commonKeys), t, opts)
+	}
+
+	if opts.TableDialect == MarkdownTableDialect {
```

Comment:
To avoid such conditions, TableDialect must be an interface with concrete implementations for Markdown, Jira, etc..